src/main/java/app/bpartners/api/service/invoice/ReferenceGenerator.java

```diff
@@ -0,0 +1,14 @@
+package app.bpartners.api.service.invoice;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class ReferenceGenerator {
+  public static String generateReference() {
```

Comment:
Do not make it static.

Instead use a class Supplier<String> 

Add a test class